### PR TITLE
Fetch transactions

### PR DESCRIPTION
**Pull request by suriksarkisyan**

### CODE REVIEW

File: Sources/Swift/StoreKit2Service.swift [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    for (_, transactions) in transactions {
+      if transactions.count > 1 {
+        var previousHandledProductId = ""
+        
+        for transaction in transactions {
+          if previousHandledProductId != transaction.productID {
+            result.append(transaction)
+            previousHandledProductId = transaction.productID
+          }
+        }
```
Comment (SpertsyanKM) [inline Sources/Swift/StoreKit2Service.swift:91]: Are we sure that transactions are sorted by product id? I mean, is it possible, that the third and the first transactions have the same product id and the second - another one? In that case, all three transactions would be added, while we expect only the first two.

File: Sources/Swift/StoreKit2Service.swift [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      resultMap[transaction.originalID] = transactionsByOriginalId
+    }
+    
+    return resultMap
+  }
+  
+  func filterGroupedTransactions(_ transactions: [UInt64: [Transaction]]) -> [Transaction] {
+    var result: [Transaction] = []
+    for (_, transactions) in transactions {
+      if transactions.count > 1 {
```
Comment (SpertsyanKM) [inline Sources/Swift/StoreKit2Service.swift:83]: I think this is an extra condition as the `if` body will work correctly for single-element arrays too.

Comment (suriksarkisyan) [reply Sources/Swift/StoreKit2Service.swift:83]: I'll fix all of these comments in the next PR because of the many project structure changes there.

**Comment by suriksarkisyan** [comment]
I'll fix all of these comments in the next PR because of the many project structure changes there.

**Pull request merged**